pkg/nodediscovery/localnodeconfig.go

```diff
@@ -35,11 +35,12 @@ func NewLocalNodeConfig(mtu mtu.MTU, config *option.DaemonConfig) (datapath.Loca
 	}
 
 	return datapath.LocalNodeConfiguration{
-		MtuConfig:               mtu,
-		EnableIPv4:              config.EnableIPv4,
-		EnableIPv6:              config.EnableIPv6,
-		EnableEncapsulation:     config.TunnelingEnabled(),
-		EnableAutoDirectRouting: config.EnableAutoDirectRouting,
+		MtuConfig:                          mtu,
+		EnableIPv4:                         config.EnableIPv4,
+		EnableIPv6:                         config.EnableIPv6,
+		EnableEncapsulation:                config.TunnelingEnabled(),
+		EnableAutoDirectRouting:            config.EnableAutoDirectRouting,
+		EnableDirectRoutingSkipUnreachable: config.EnableDirectRoutingSkipUnreachable,
```

Comment:
If I traced all the relevant codepaths, `EnableDirectRoutingSkipUnreachable` is dependent upon `EnableAutoDirectRouting`. Since the name of this flag starts with "EnableDirectRouting", it might imply that enabling it enables direct routing, which is not the case. Could you make these changes?

* Rename the flag to `DirectRoutingSkipUnreachable`
* Add some validation in this function (similar to the CIDR validation above), which returns an error if this flag is enabled and `EnableAutoDirectRouting` is not enabled.

Please let me know if you see a case where this flag is not dependent.